src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -76,18 +76,20 @@ public void preProcess(SearchContext context) {
             Aggregator[] aggregators;
             try {
                 aggregators = context.aggregations().factories().createTopLevelAggregators(aggregationContext);
+                for (int i = 0; i < aggregators.length; i++) {
+                    if (aggregators[i] instanceof GlobalAggregator == false) {
+                        collectors.add(aggregators[i]);
+                    }
+                }
+                context.aggregations().aggregators(aggregators);
+                final BucketCollector collector = BucketCollector.wrap(collectors);
+                collector.preCollection();
```

Comment:
Can the these two lines not be put inside of the if statement so we don't create the wrapper unless we are going to use it?